scipy/stats/_multivariate.py

```diff
@@ -5685,3 +5687,494 @@ def _sample_uniform_direction(dim, size, random_state):
     samples = random_state.standard_normal(samples_shape)
     samples /= np.linalg.norm(samples, axis=-1, keepdims=True)
     return samples
+
+
+_vonmises_fisher_doc_default_callparams = """\
+mu : array_like
+    Mean direction of the distribution. Must be a one-dimensional unit
+    vector of norm 1.
+kappa : float
+    Concentration parameter. Must be positive."""
+
+_vonmises_fisher_doc_callparams_note = "bla"
+
+_vonmises_fisher_doc_frozen_callparams = ""
+
+_vonmises_fisher_doc_frozen_callparams_note = """\
+See class definition for a detailed description of parameters."""
+
+vonmises_fisher_docdict_params = {
+    '_vonmises_fisher_doc_default_callparams':
+        _vonmises_fisher_doc_default_callparams,
+    '_vonmises_fisher_doc_callparams_note':
+        _vonmises_fisher_doc_callparams_note,
+    '_doc_random_state': _doc_random_state
+}
+
+vonmises_fisher_docdict_noparams = {
+    '_vonmises_fisher_doc_default_callparams':
+        _vonmises_fisher_doc_frozen_callparams,
+    '_vonmises_fisher_doc_callparams_note':
+        _vonmises_fisher_doc_frozen_callparams_note,
+    '_doc_random_state': _doc_random_state
+}
+
+
+class vonmises_fisher_gen(multi_rv_generic):
+    r"""A von Mises-Fisher variable.
+
+    The `mu` keyword specifies the mean direction vector. The `kappa` keyword
+    specifies the concentration parameter.
+
+    Methods
+    -------
+    pdf(x, mu=None, kappa=1)
+        Probability density function.
+    logpdf(x, mu=None, kappa=1)
+        Log of the probability density function.
+    rvs(mu=None, kappa=1, size=1, random_state=None)
+        Draw random samples from a von Mises-Fisher distribution.
+    entropy()
+        Compute the differential entropy of the von Mises-Fisher distribution.
+    fit(data)
+        Fit a von Mises-Fisher distribution to data.
+
+    Parameters
+    ----------
+    %(_vonmises_fisher_doc_default_callparams)s
+    %(_doc_random_state)s
+
+    See Also
+    --------
+    scipy.stats.vonmises : Von-Mises Fisher distribution in 2D on a circle
+
+    Notes
+    -----
+    %(_vonmises_fisher_doc_callparams_note)s
+
+    The von Mises-Fisher distribution is a directional distribution on the
+    n-dimensional unit sphere. The probability density function is
+
+    .. math::
+
+        f(\mathbf{x}) = \frac{\kappa^{d/2-1}}{(2\pi)^{d/2}I_{d/2-1}(\kappa)}
+               \exp\left(\kappa \mathbf{\mu}^T\mathbf{x}\right),
+
+    where :math:`\mu` is the mean direction, :math:`\kappa` the
```

Comment:
I think we should add more details around kappa. It's not clear what it does and what range can be used. Would be a good example.